# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: docshell/base/CanonicalBrowsingContext.cpp
# Commit: 07c8392cb04d
# Full Hash: 07c8392cb04d095caa8e65ee1a10ccc5f7352eb6
# Author: Andrew McCreight <continuation@gmail.com>
# Date: 2021-05-07 21:46:25
# Regressor Bug: 1618547
# File Overlap Count: 1
# Description:
#   Bug 1618547 - Support Fission in the process priority manager. r=kmag,gsvelto
#   
#   With Fission, there can be multiple BrowserParents in a single tab, so
#   this patch moves the tracking of active tabs onto the top BrowsingContext
#   in a tab. If the priority of a top BC is changed, then the activity
# ==============================================================================

diff -r 285861b32934 -r 07c8392cb04d docshell/base/CanonicalBrowsingContext.cpp
--- a/docshell/base/CanonicalBrowsingContext.cpp	Fri May 07 17:42:21 2021 +0000
+++ b/docshell/base/CanonicalBrowsingContext.cpp	Fri May 07 17:51:34 2021 +0000
@@ -223,6 +223,11 @@
     BackgroundSessionStorageManager::PropagateManager(Id(), aNewContext->Id());
   }
 
+  // Transfer the ownership of the priority active status from the old context
+  // to the new context.
+  aNewContext->mPriorityActive = mPriorityActive;
+  mPriorityActive = false;
+
   MOZ_ASSERT(aNewContext->mLoadingEntries.IsEmpty());
   mLoadingEntries.SwapElements(aNewContext->mLoadingEntries);
   MOZ_ASSERT(!aNewContext->mActiveEntry);